content/open-source/wolfi/wolfi-with-dockerfiles.md

```diff
@@ -0,0 +1,131 @@
+
+The following command will clone the demos repository in your `/tmp` folder:
+
+```shell
+mkdir /tmp/images-demos &&  \
+  git clone https://github.com/chainguard-dev/edu-images-demos.git \
+  /tmp/images-demos
+```
+
+We'll now copy the demo application to a location inside your home folder.
+
+```shell
+mkdir ~/inky && cp -R /tmp/images-demos/python/inky/* ~/inky/
+```
+
+You can now enter the newly created directory in your home folder and inspect its contents:
+
+```shell
+cd ~/inky && ls -la
+```
+
+This application will take in an image (`inky.png`) file and convert it to ANSI escape sequences to render it on the CLI. The code is on the `inky.py` file, while the `requirements.txt` file has the dependencies required by the application: [setuptools](https://pypi.org/project/setuptools/) and [climage](https://pypi.org/project/climage/).
+
+For your reference, here is the complete `inky.py` script:
+
+```python
+'''import climage module to display images on terminal'''
+import climage
+
+def main():
+    '''Take in PNG and output as ANSI to temrinal'''
+    output = climage.convert('inky.png', is_unicode=True)
+    print(output)
+
+if __name__ == "__main__":
+    main()
+```
+
+You'll notice that there's already a Dockerfile in that directory, but it uses the [Python Chainguard image](https://edu.chainguard.dev/chainguard/chainguard-images/reference/python/overview/) in a multi-stage build. In the next step, we'll replace that with a new Dockerfile that uses the [Wolfi-base](https://edu.chainguard.dev/chainguard/chainguard-images/reference/wolfi-base/image_specs/) image to build a Python image from scratch, using Wolfi apks.
+
+To test that the application works as expected, first install the dependencies using Pip.
+
+```shell
+pip3 install -r requirements.txt
+```
+
+Then, run the application with:
+
+```shell
+python3 inky.py
+```
+
+You’ll receive a representation of the Chainguard Inky logo on the command line. With the demo application ready, you can now move on to building the Wolfi container image.
+
+## Creating the Dockerfile
+
+Now we'll create the Dockerfile to run the application. This Dockerfile will set up a new WORKDIR, copy relevant files, and install dependencies with Pip. It will also define the entry point that will be executed when we run this image with `docker run`.
+
+You can rename the old Dockerfile if you want to keep it for tests later.
+
+```shell
+mv Dockerfile _DockerfileBkp
+```
+
+Then, create a new Dockerfile:
+
+```shell
+nano Dockerfile
+```
+
+Copy the following content to it:
+
+```
+FROM cgr.dev/chainguard/wolfi-base
+
+WORKDIR /app
+COPY requirements.txt inky.py inky.png /app/
+RUN apk add python-3.11 py3.11-pip && \
+    pip3 install -r requirements.txt --user
+
+ENTRYPOINT ["python", "/app/inky.py"]
+
+```
+
+After saving the file, you can build your application runtime image with:
+
+```shell
+docker build . -t inky-demo
```

Comment:
I needed to run `sudo` here on macOS, and you may want to remind macOS users that they need to have Docker running